pkg/engine/variables/vars_test.go

```diff
@@ -1157,3 +1157,48 @@ func Test_getJMESPath(t *testing.T) {
 	assert.Equal(t, "spec.containers[0].volumes[1]", getJMESPath("/validate/pattern/spec/containers/0/volumes/1"))
 	assert.Equal(t, "[0]", getJMESPath("/mutate/overlay/0"))
 }
+
+func Test_SubstituteSuccess1(t *testing.T) {
+	ctx := context.NewContext()
+
+	var resourceRaw = []byte(`
+    {
+        "spec":{
+               "containers":[
+                {
+
+                    "image":"nginx:1.19",
+                    "name":"nginx",
+                    "resources":{
+                        "limits":{
+                            "cpu":"500m",
```

Comment:
What happens if e.g. `cpu` is set to `1`? If I understand the current implementation correctly, we would end up with an invalid request value for cpu

Reply:
> What happens if e.g. `cpu` is set to `1`? If I understand the current implementation correctly, we would end up with an invalid request value for cpu

There's something wrong here. I check it

Reply:
> What happens if e.g. `cpu` is set to `1`? If I understand the current implementation correctly, we would end up with an invalid request value for cpu

**fixed**

- ClusterPolicy
```
apiVersion : kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: mutate-update-requests-limits
spec:
  rules:
  - name: mutate-update-requests-limits
    match:
      resources:
        kinds:
        - Pod
        namespaces:
        - default
    mutate:
      foreach:
      - list: "request.object.spec.containers"
        patchStrategicMerge:
          spec:
            containers:
            - name: "{{ element.name }}"
              resources:
                requests:
                  cpu: "{{ element.resources.limits.cpu / 2 }}"
                  memory: "{{ element.resources.limits.memory / 2 }}
```

- The source manifest
```
apiVersion: v1
kind: Pod
metadata:
  name: kyverno-test
  namespace: default
spec:
  containers:
  - name: nginx
    image: nginx:1.19
    command: ["/bin/bash", "-c", "sleep INF"]
    resources:
      requests:
        memory: "128Mi"
        cpu: "500m"
      limits:
        memory: "129Mi"
        cpu: 1
```
- The execution result
```
spec:
  containers:
  - command:
    - /bin/bash
    - -c
    - sleep INF
    image: nginx:1.19
    imagePullPolicy: IfNotPresent
    name: nginx
    resources:
      limits:
        cpu: "1"
        memory: 129Mi
      requests:
        cpu: 500m
        memory: 65Mi
```

**Note that if mem is a decimal, then we returns the least integer value greater than  mem.**

Reply:
Using the JMESPath custom functions
```
apiVersion : kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: mutate-update-requests-limits
spec:
  rules:
  - name: mutate-update-requests-limits
    match:
      resources:
        kinds:
        - Pod
        namespaces:
        - default
    mutate:
      foreach:
      - list: "request.object.spec.containers"
        patchStrategicMerge:
          spec:
            containers:
            - name: "{{ element.name }}"
              resources:
                requests:
                  cpu: "divide(`{{ element.resources.limits.cpu }}`, `2`)"
                  memory: "divide(`{{ element.resources.limits.memory}}`, `2`)"
```